main_desktop.py

```
@@ -360,23 +379,50 @@ def _save_selection_for_project(self, directory_path, selected_files):
             
-            # Sauvegarder
-            with open(SELECTION_CACHE_PATH, 'w', encoding='utf-8') as f:
-                json.dump(cache, f, indent=2, ensure_ascii=False)
+            # Remplacer atomiquement l'ancien fichier
+            os.replace(tmppath, SELECTION_CACHE_PATH)
+            
+            # Définir les permissions strictes (lecture/écriture propriétaire uniquement)
+            try:
+                os.chmod(SELECTION_CACHE_PATH, 0o600)
+            except:
+                pass  # Sur Windows, chmod peut ne pas fonctionner
```

Ruff (0.12.2)

Use `contextlib.suppress(BaseException)` instead of `try`-`except`-`pass`

(SIM105)